keyboards/annepro2/qmk_ap2_led.c

```diff
@@ -1,115 +1,106 @@
+#include <string.h>
+#include <stdio.h>
 #include "hal.h"
 #include "annepro2.h"
 #include "qmk_ap2_led.h"
+#include "protocol.h"
 
-bool AP2_LED_ENABLED = false;
-bool AP2_LED_DYNAMIC_PROFILE = false;
-bool AP2_FOREGROUND_COLOR_SET = false;
+annepro2Led_t       ledMask[KEY_COUNT];
+annepro2LedStatus_t annepro2LedStatus;
 
-void annepro2LedDisable(void)
-{
-  sdPut(&SD0, CMD_LED_OFF);
-  AP2_LED_ENABLED = false;
-}
+void ledCommandCallback(const message_t *msg) {
+    switch (msg->command) {
+        case CMD_LED_STATUS:
+            annepro2LedStatus.amountOfProfiles = msg->payload[0];
+            annepro2LedStatus.currentProfile   = msg->payload[1];
+            annepro2LedStatus.matrixEnabled    = msg->payload[2];
+            annepro2LedStatus.isReactive       = msg->payload[3];
+            annepro2LedStatus.ledIntensity     = msg->payload[4];
+            annepro2LedStatus.errors           = msg->payload[5];
+            break;
 
-void annepro2LedEnable(void)
-{
-  sdPut(&SD0, CMD_LED_ON);
-  AP2_LED_ENABLED = true;
+#ifdef CONSOLE_ENABLE
+        case CMD_LED_DEBUG:
+            /* TODO: Don't use printf. */
+            printf("LED:");
```

Comment:
With %02x the printf makes some sense; but this could use some low-level write(), but I don't know the API.

Reply:
it is fine. printf is not slow for hexes. It can be done with shift and mask without software division which is fine. There is no need to reimplement those subroutines.